hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMDatanodeProtocolServer.java

```diff
@@ -115,14 +115,15 @@
    */
   private final RPC.Server datanodeRpcServer;
 
-  private final StorageContainerManager scm;
+  private final StorageContainerManagerInterface scm;
```

Comment:
It's uncanny to call something an interface, why not use IDE to refactor the StorageContainerManager into and interface? It would not be too big a change for callers.

Reply:
I can change the name of the interface. I just did not want to move everything to an interface now, since we are not sure what all we will need to override. At the end of this, we can expect the interface to have everything Recon needs to override.

Reply:
But the point is after your patch, anybody changing SCM will change the interface and thereby Recon will get the change too. It's a good thing that way. From Recon point of view, empty method behavior is also ok until we use those methods.

Reply:
Done.